### PR TITLE
Convert simple_embedding_test to run_binary_test

### PR DESCRIPTION
**Pull request by hcindyl**
Split the TEST_P into two tests so cmake can apply the right condition.
Supports both bazel test and ctest.

### CODE REVIEW

**Comment by hcindyl** [comment]
This is the follow up of https://github.com/google/iree/pull/5524 and the last TODOs for https://github.com/google/iree/issues/5481

File: iree/samples/simple_embedding/CMakeLists.txt [hunk 1]
```diff
@@ -66,26 +66,30 @@ if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64")
   return()
 endif()
 
-iree_cc_test(
+if(${IREE_TARGET_BACKEND_DYLIB-LLVM-AOT} AND ${IREE_HAL_DRIVER_DYLIB})
```
Comment (ScottTodd) [inline iree/samples/simple_embedding/CMakeLists.txt:69]: this condition is already checked by 

```
if(NOT ${IREE_TARGET_BACKEND_DYLIB-LLVM-AOT} OR NOT ${IREE_TARGET_BACKEND_VULKAN-SPIRV}
   OR NOT ${IREE_HAL_DRIVER_DYLIB} OR NOT ${IREE_HAL_DRIVER_VULKAN})
  return()
endif()
```

above. I think it's fine to keep or omit these redundant checks. As we get further into multiarch/heterogeneous execution, we can change "simple_embedding" to just use a single backend, then add a separate sample showing those features.

**Review by ScottTodd — COMMENTED** [review]
Not sure what's up with the Kokoro CI... it hasn't started running builds/tests yet?

File: iree/samples/simple_embedding/CMakeLists.txt (hunk 1 above)
Comment (hcindyl) [reply iree/samples/simple_embedding/CMakeLists.txt]: Done.

**Review by ScottTodd — APPROVED** [review]
Kokoro still isn't running builds/tests, but this LGTM.

**Comment by ScottTodd** [comment]
> Kokoro still isn't running builds/tests, but this LGTM.

Oooh, it finally started running :D so LGTM once build results come back.

**Pull request merged**